RecoTracker/TkTrackingRegions/src/GlobalTrackingRegion.cc

```diff
@@ -90,28 +88,51 @@ std::unique_ptr<HitRZCompatibility> GlobalTrackingRegion::checkRZ(const DetLayer
   SimpleLineRZ rightLine(vtxR, outerR);
   HitRZConstraint rzConstraint(leftLine, rightLine);
 
-  if
-    UNLIKELY(theUseMS) {
-      MultipleScatteringParametrisation iSigma(layer, iSetup);
-      PixelRecoPointRZ vtxMean(0., origin().z());
+  if UNLIKELY (theUseMS) {
+    MultipleScatteringParametrisation iSigma(layer, iSetup);
+    PixelRecoPointRZ vtxMean(0., origin().z());
 
-      float innerScatt = 3.f * (outerlayer ? iSigma(ptMin(), vtxMean, outerred, outerlayer->seqNum())
-                                           : iSigma(ptMin(), vtxMean, outerred));
+    float innerScatt = 3.f * (outerlayer ? iSigma(ptMin(), vtxMean, outerred, outerlayer->seqNum())
+                                         : iSigma(ptMin(), vtxMean, outerred));
 
-      float cotTheta = SimpleLineRZ(vtxMean, outerred).cotLine();
+    float cotTheta = SimpleLineRZ(vtxMean, outerred).cotLine();
 
-      if (isBarrel) {
-        float sinTheta = 1 / std::sqrt(1 + sqr(cotTheta));
-        corr = innerScatt / sinTheta + dz;
-      } else {
-        float cosTheta = 1 / std::sqrt(1 + sqr(1 / cotTheta));
-        corr = innerScatt / cosTheta + dr;
-      }
+    if (isBarrel) {
+      float sinTheta = 1 / std::sqrt(1 + sqr(cotTheta));
+      corr = innerScatt / sinTheta + dz;
+    } else {
+      float cosTheta = 1 / std::sqrt(1 + sqr(1 / cotTheta));
+      corr = innerScatt / cosTheta + dr;
     }
+  }
 
   if (isBarrel) {
     return std::make_unique<HitZCheck>(rzConstraint, HitZCheck::Margin(corr, corr));
   } else {
     return std::make_unique<HitRCheck>(rzConstraint, HitRCheck::Margin(corr, corr));
   }
 }
+
+void GlobalTrackingRegion::checkTracks(reco::TrackCollection const& tracks, std::vector<bool>& mask) const {
+  math::XYZPoint regOrigin(origin().x(), origin().y(), origin().z());
```

Comment:
make it const

Reply:
does it make any difference ?

Reply:
I haven't checked if the generated code is different, and I don't expect it would be. However, my understanding is that we can and should be liberal with `const` in signaling intent to the compiler. Let me know if I have misunderstood. Thanks!

Reply:
Adding `const` clearly doesn't harm - but as you say, I don't expect to be any difference in the generated code, since in simple cases like this the compiler should figure out that the variable(s) are not being modified.
I'd say it's more about signalling intention to the developers, which of course doesn't harm...